### PR TITLE
Front end search form

### PR DESCRIPTION
**Pull request by CressChe**
https://trello.com/c/eOfaAo1W/57-front-end-search-form

- no reset button yet
- no front end test 
(we will work on these tonight)

### CODE REVIEW

File: WhaleSpotting/Services/SightingsService.cs [hunk 1]
```diff
@@ -46,13 +46,23 @@ public async Task<List<SightingResponseModel>> GetAllSightings()
...
                 .Where(s => searchSighting.SightedTo == null || s.SightedAt <= searchSighting.SightedTo)
-                .Where(s => string.IsNullOrEmpty(searchSighting.OrcaPod) || s.OrcaPod == searchSighting.OrcaPod)
-                .Where(s => string.IsNullOrEmpty(searchSighting.Location) || s.Location == searchSighting.Location)
-                .Where(s => searchSighting.Confirmed == null || s.Confirmed == searchSighting.Confirmed)
+                .Where(s => searchSighting.Latitude == null || s.Latitude > lowerLatitude && s.Latitude < upperLatitude)
+                .Where(s => searchSighting.Longitude == null || s.Longitude > lowerLongitude && s.Longitude < upperLongitude)
+                .Where(s => searchSighting.OrcaType == null || s.OrcaType == searchSighting.OrcaType)
+                .Where(s => string.IsNullOrEmpty(searchSighting.OrcaPod) || s.OrcaPod.ToLower() == searchSighting.OrcaPod.ToLower())
+                .Where(s => string.IsNullOrEmpty(searchSighting.Location) || s.Location.ToLower() == searchSighting.Location.ToLower())
+                .Where(s => s.Confirmed == true)
```
Comment (oskarwilliams) [inline WhaleSpotting/Services/SightingsService.cs]:
```suggestion
                .Where(s => (s.Confirmed && searchSighting.Confirmed != false) || (!s.Confirmed && searchSighting.Confirmed == false))
```

File: WhaleSpotting/Services/SightingsService.cs [hunk 2]
```diff
@@ -46,13 +46,23 @@ public async Task<List<SightingResponseModel>> GetAllSightings()
...
+            var lowerLatitude = searchSighting.Latitude - radiusInCoords;
+
+            var upperLongitude = searchSighting.Longitude + radiusInCoords;
+            var lowerLongitude = searchSighting.Longitude - radiusInCoords;
+
             var sightings = await _context.Sightings
                 .Where(s => searchSighting.Species == null || s.Species == searchSighting.Species)
-                .Where(s => searchSighting.SightedFrom == null || s.SightedAt >= searchSighting.SightedFrom)
+                .Where(s => searchSighting.SightedFrom == null || s.SightedAt >= Convert.ToDateTime(searchSighting.SightedFrom))
                 .Where(s => searchSighting.SightedTo == null || s.SightedAt <= searchSighting.SightedTo)
```
Comment (oskarwilliams) [inline WhaleSpotting/Services/SightingsService.cs:60]: This should probably convert to date tiome as well?

File: WhaleSpotting/Models/RequestModels/SearchSightingRequestModel.cs [hunk 3]
```diff
@@ -26,6 +26,8 @@ public class SearchSightingRequestModel
         public OrcaType? OrcaType { get; set; }
 
         public string OrcaPod { get; set; } = null;
-        public bool? Confirmed { get; set; }
+        
+        [Range(0, int.MaxValue, ErrorMessage = "Radius in km must be greater than 0")]
+        public int? RadiusKm { get; set; } = 50;
```
Comment (oskarwilliams) [inline WhaleSpotting/Models/RequestModels/SearchSightingRequestModel.cs:31]: Add the confirmed back in pls

File: WhaleSpotting/react-app/src/styles/Home.scss [hunk 4]
```diff
@@ -22,6 +22,65 @@
...
+
+        .input-field {
+            height: 85%;
+            width: 100%;
+            padding-left: 10px;
+            border-radius: 5px;
+            transition: all 0.3s ease;
+            border: 1px solid $primaryColour;
+            box-sizing: border-box;
+
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/styles/Home.scss:52]: extra new line

**Review by oskarwilliams — CHANGES_REQUESTED** [review]

File: WhaleSpotting/Services/SightingsService.cs (hunk 2 above)
Comment (francescastocco) [reply WhaleSpotting/Services/SightingsService.cs:60]: Not required for either

File: WhaleSpotting/Services/SightingsService.cs (hunk 1 above)
Comment (francescastocco) [reply WhaleSpotting/Services/SightingsService.cs]: reverted to s => searchSighting.Confirmed == null || s.Confirmed == searchSighting.Confirmed

**Review by oskarwilliams — APPROVED** [review]

**Pull request merged**